joblib/_multiprocessing_helpers.py

```diff
@@ -22,18 +24,34 @@
 #            issue a warning if not
 if mp is not None:
     try:
-        # Use the spawn context
-        if sys.version_info < (3, 3):
-            Semaphore = mp.Semaphore
+        if sys.platform == "win32":
+            # Use the spawn context
+            if PY27:
+                Semaphore = mp.Semaphore
+            else:
+                # Using mp.Semaphore has a border effect and set the default
+                # backend for multiprocessing. To avoid that, we use the
+                # 'spawn' context which is available on all supported platforms
+                ctx = mp.get_context('spawn')
+                Semaphore = ctx.Semaphore
+            _sem = Semaphore()
         else:
-            # Using mp.Semaphore has a border effect and set the default
-            # backend for multiprocessing. To avoid that, we use the 'spawn'
-            # context which is available on all supported platforms.
-            ctx = mp.get_context('spawn')
-            Semaphore = ctx.Semaphore
-        _sem = Semaphore()
-        del _sem  # cleanup
-    except (ImportError, OSError) as e:
+            # try to create a named semaphore using
```

Comment:
Maybe you could improve the comment to state we try to create the semaphore manually to avoid create a semaphore or resource tracker process just when importing joblib.

Reply:
Updated this comment too.